public/src/managers/ServerProvider.ts

```diff
@@ -309,28 +311,25 @@ export class ServerProvider implements IUserProvider, ICourseProvider {
         user.setIsadmin(promote);
         const result = await this.grpcHelper.updateUser(user);
         // we are not interested in user data returned in this case, only checking that there were no errors
-        return result.status.getCode() === 0;
+        return this.responseCodeSuccess(result);
     }
 
     public async updateUser(user: User): Promise<boolean> {
         const result = await this.grpcHelper.updateUser(user);
-        if (result.status.getCode() !== 0 || !result.data) {
-            return false;
-        }
-        return true;
+        return result.data ? this.responseCodeSuccess(result) : false;
```

Comment:
This pattern matches my proposed change of the `responseCodeSuccess`. This would become:
```typescript
return this.isSuccess(result);
```
Naming suggestions:
`isSuccess` or `responseSuccess` or something better...